game/procurement.py

```diff
@@ -185,6 +185,13 @@ def fulfill_aircraft_request(
     ) -> Tuple[float, bool]:
         for squadron in squadrons:
             price = squadron.aircraft.price * quantity
+            # Final check to make sure the number of aircraft won't exceed the number of available pilots
+            # after fulfilling this aircraft request.
+            if (
+                squadron.pilot_limits_enabled
+                and squadron.expected_size_next_turn + quantity > squadron.max_size
```

Comment:
And now with the clarity of a much smaller PR, I see the bug :) `squadron.max_size` is the maximum number of pilots. You want the number of pilots that are expected to be present next turn. Add this to `Squadron` and use that:

```python
@property
def expected_pilots_next_turn(self) -> int:
    return len(self.active_pilots) + self.settings.squadron_replenishment_rate
```

Reply:
This is now done, marking as ready for review.